code/modules/mob/living/status_procs.dm

```diff
@@ -67,8 +67,12 @@
 #define SET_STATUS_EFFECT_STRENGTH(T, A) \
 	A = max(A, 0);\
 	if(A) {;\
-		var/datum/status_effect/transient/S = has_status_effect(T) || apply_status_effect(T);\
-		S.strength = A;\
+		var/datum/status_effect/transient/S = has_status_effect(T);\
+		if(!S) {;\
+			apply_status_effect(T, A);\
+		} else {;\
+			S.strength = A;\
+		};\
```

Comment:
changed this so that the things that check for the amount of strength `on_apply` actually work. with the prior implementation, if you checked for the strength `on_apply` it would give you 0.